common/pipeline_storage/src/test/scala/uk/ac/wellcome/pipeline_storage/WorkIndexableTest.scala

```diff
@@ -132,7 +131,30 @@ class WorkIndexerTest
       }
     }
 
-    it("overrides a identified Work with invisible work with higher version") {
+    it("overrides a merged work with one that has been merged again") {
+      val mergedWork1 = identifiedWork(nMergedSources = 1)
+      val mergedWork2 = identifiedWork(nMergedSources = 2)
+      val mergedWork3 = identifiedWork(nMergedSources = 3)
+
+      withWorksIndexAndIndexer {
+        case (index, indexer) =>
+          val insertFuture = ingestInOrder(indexer)(
+            mergedWork1,
+            mergedWork3,
+            mergedWork2
+          )
+          whenReady(insertFuture) { result =>
+            assertIngestedWorkIs(
+              result = result,
+              ingestedWork = mergedWork3,
+              index = index
+            )
+          }
+      }
+    }
```

Comment:
It looks to me like this test will create three completely independent works, all with different canonical IDs. That doesn't seem right.

Reply:
Oops! Think I lost this in a merge, will fix